contracts/src/LightClient.sol

```diff
@@ -199,34 +203,38 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     /// @dev Initialization of contract variables happens in this method because the LightClient
     /// contract is upgradable and thus has its constructor method disabled.
     /// @param _genesis The initial state of the light client
+    /// @param _genesisStakeState The initial stake state of the light client
     /// @param _stateHistoryRetentionPeriod The maximum retention period (in seconds) for the state
     /// history
-    function _initializeState(LightClientState memory _genesis, uint32 _stateHistoryRetentionPeriod)
-        internal
-    {
+    function _initializeState(
+        LightClientState memory _genesis,
+        StakeState memory _genesisStakeState,
+        uint32 _stateHistoryRetentionPeriod
+    ) internal {
         // stake table commitments and threshold cannot be zero, otherwise it's impossible to
         // generate valid proof to move finalized state forward.
         // Whereas blockCommRoot can be zero, if we use special value zero to denote empty tree.
         // feeLedgerComm can be zero, if we optionally support fee ledger yet.
         if (
             _genesis.viewNum != 0 || _genesis.blockHeight != 0
-                || BN254.ScalarField.unwrap(_genesis.stakeTableBlsKeyComm) == 0
-                || BN254.ScalarField.unwrap(_genesis.stakeTableSchnorrKeyComm) == 0
-                || BN254.ScalarField.unwrap(_genesis.stakeTableAmountComm) == 0
-                || _genesis.threshold == 0
+                || BN254.ScalarField.unwrap(_genesisStakeState.stakeTableBlsKeyComm) == 0
+                || BN254.ScalarField.unwrap(_genesisStakeState.stakeTableSchnorrKeyComm) == 0
+                || BN254.ScalarField.unwrap(_genesisStakeState.stakeTableAmountComm) == 0
+                || _genesisStakeState.threshold == 0
         ) {
             revert InvalidArgs();
         }
-        states[genesisState] = _genesis;
-        states[finalizedState] = _genesis;
+        genesisState = _genesis;
+        genesisStakeState = _genesisStakeState;
+        finalizedState = _genesis;
 
         stateHistoryRetentionPeriod = _stateHistoryRetentionPeriod;
 
-        bytes32 initStakeTableComm = computeStakeTableComm(_genesis);
+        bytes32 initStakeTableComm = computeStakeTableComm(_genesisStakeState);
         votingStakeTableCommitment = initStakeTableComm;
-        votingThreshold = _genesis.threshold;
+        votingThreshold = _genesisStakeState.threshold;
         frozenStakeTableCommitment = initStakeTableComm;
-        frozenThreshold = _genesis.threshold;
+        frozenThreshold = _genesisStakeState.threshold;
```

Comment:
imo, we should remove all of these: 
```solidity
bytes32 public votingStakeTableCommitment;
uint256 public votingThreshold;
bytes32 public frozenStakeTableCommitment;
uint256 public frozenThreshold;
mapping(uint32 index => LightClientState value) public states;

function computeStakeTableComm(LightClientState memory state) returns (bytes32);
```

i roughly explain the "frozen/voting stake table" here: https://www.notion.so/espressosys/Light-Client-Contract-Details-6c396f7824e34356bdcda710015291f9?pvs=4#57affaec4dbf4508a27b169255fde936

TL;DR is `votingXX` is the stake table snapshot/version that we will be using when validating quorum signatures (or SNARK that proves a valid QC)

but since we are not changing them anyway, just remove them for now and use `genesisState` you introduced.

Reply:
addressed in [c63635e](https://github.com/EspressoSystems/espresso-sequencer/pull/1939/commits/c63635e04eb70c651e8bd2d3bf7295e1ee5f9b80)